activerecord/lib/active_record/core.rb

```diff
@@ -25,6 +25,12 @@ module Core
       # Specifies the job used to destroy associations in the background
       class_attribute :destroy_association_async_job, instance_writer: false, instance_predicate: false, default: false
 
+      ##
+      # :singleton-method:
+      #
+      # Specifies the number of associated records destroyed in a single background job.
+      class_attribute :destroy_association_async_batch_size, instance_writer: false, instance_predicate: false, default: false
```

Comment:
(non-blocking nitpick) To me `nil` would make more sense than `false` as a default here

Reply:
Agreed! Made `nil` the default.